### ISSUE TITLE
Which is the actual way to store the Adapter after PEFT finetuning

### ISSUE THREAD

**Issue by pradeepdev-1995** [facebookresearch/llama#986]
I am finetuning the mistral model using the following configurations
```
training_arguments = TrainingArguments(
    output_dir=output_dir,
    per_device_train_batch_size=per_device_train_batch_size,
    gradient_accumulation_steps=gradient_accumulation_steps,
    optim=optim,
    save_steps=save_steps,
    logging_strategy="steps",
    logging_steps=10,
    learning_rate=learning_rate,
    weight_decay=weight_decay,
    fp16=fp16,
    bf16=bf16,
    max_grad_norm=max_grad_norm,
    max_steps=13000,
    warmup_ratio=warmup_ratio,
    group_by_length=group_by_length,
    lr_scheduler_type=lr_scheduler_type
)
trainer = SFTTrainer(
    model=peft_model,
    train_dataset=data,
    peft_config=peft_config,
    dataset_text_field=" column name",
    max_seq_length=3000,
    tokenizer=tokenizer,
    args=training_arguments,
    packing=packing,
)
trainer.train()
```

during this training I am getting the multiple checkpoints in the specified output directory ```output_dir```.

Once the model training is over I can save the model using 
```
trainer.save_model()
```
Not only that i can save the final model using
```
trainer.model.save_pretrained("path")
```

So I bit confused. Which is the actual way to store the adapter after PEFT based lora fine-tuning

whether it is
1 - Take the least loss checkpoint folder from the ```output_dir```
or
2 - save the adapter using 
```
trainer.save_model()
```
or
3 - this method
```
trainer.model.save_pretrained("path")
```

**Comment by subramen** [comment]
Hi @pradeepdev-1995 you should ask this question on the PEFT / TRL repos as this is relevant to that API and not the llama model.

But IIRC I think you can use `peft_model.save_pretrained("adapter_path")` to save the adapter and then load it like `model = PeftModel(base_model, PeftConfig.from_pretrained(adapter_path))`

cc @pacman100 for PEFT expertise

**Comment by pacman100** [comment]
Hello, 

Thank you @subramen for the ping.

@pradeepdev-1995, `trainer.save_model()` should suffice to save the adapters when using Trainer and PEFT. This will save the model at the end of the training loop.
